src/proto/media/partition.rs

```diff
@@ -0,0 +1,202 @@
+    boot_indicator: u8,
+    starting_chs: [u8; 3],
+    os_type: MbrOsType,
+    ending_chs: [u8; 3],
+    starting_lba: u32,
+    size_in_lba: u32,
+}
+
+impl MbrPartitionRecord {
+    /// True if the partition is a bootable legacy partition.
+    pub fn is_bootable(&self) -> bool {
+        self.boot_indicator == 0x80
+    }
+
+    /// Start of the partition in CHS address format.
+    pub fn starting_chs(&self) -> [u8; 3] {
+        self.starting_chs
+    }
+
+    /// End of the partition in CHS address format.
+    pub fn ending_chs(&self) -> [u8; 3] {
+        self.ending_chs
+    }
+
+    /// Type of partition.
+    pub fn os_type(&self) -> MbrOsType {
+        self.os_type
+    }
+
+    /// Starting LBA of the partition on the disk.
+    pub fn starting_lba(&self) -> u32 {
+        self.starting_lba
+    }
+
+    /// Size of the partition in LBA units of logical blocks.
+    pub fn size_in_lba(&self) -> u32 {
+        self.size_in_lba
+    }
+}
+
+/// GPT partition entry.
+///
+/// UEFI specification: table 5-6.
+#[repr(C)]
+#[repr(packed)]
+#[derive(Clone, Copy, Debug)]
+pub struct GptPartitionEntry {
+    partition_type_guid: Guid,
+    unique_partition_guid: Guid,
+    starting_lba: u64,
+    ending_lba: u64,
+    attributes: u64,
+    partition_name: [Char16; 36],
+}
+
+impl GptPartitionEntry {
+    /// GUID that defines the type of this Partition. A value of zero
+    /// indicates that this partition entry is used.
+    pub fn partition_type(&self) -> Guid {
+        self.partition_type_guid
+    }
+
+    /// GUID that is unique for every partition entry.
+    pub fn unique_partition(&self) -> Guid {
+        self.unique_partition_guid
+    }
+
+    /// Starting LBA of the partition.
+    pub fn starting_lba(&self) -> u64 {
+        self.starting_lba
+    }
+
+    /// Ending LBA of the partition.
+    pub fn ending_lba(&self) -> u64 {
+        self.ending_lba
+    }
+
+    /// All attribute bits of the partition.
+    ///
+    /// The bits are defined in table 5-8 of the UEFI specification.
+    pub fn attributes(&self) -> u64 {
+        self.attributes
+    }
+
+    /// Null-terminated string containing a human-readable name of the
+    /// partition.
+    pub fn partition_name(&self) -> [Char16; 36] {
+        self.partition_name
+    }
+}
+
+newtype_enum! {
+    /// Partition type.
+    ///
+    /// UEFI specification: section 13.18.
+    pub enum PartitionType: u32 => {
+        /// Partition is not MBR or GPT.
+        OTHER = 0x00,
+        /// MBR partition.
+        MBR = 0x01,
+        /// GPT partition.
+        GPT = 0x02,
+    }
+}
+
+#[repr(C)]
+#[derive(Clone, Copy)]
+union PartitionInfoRecord {
+    /// MBR data.
+    mbr: MbrPartitionRecord,
+
+    /// GPT data.
+    gpt: GptPartitionEntry,
+}
+
+newtype_enum! {
+    /// Partition info protocol revision.
+    ///
+    /// UEFI specification: section 13.18.
+    pub enum PartitionInfoRevision: u32 => {
+        /// Revision of the partition info protocol.
```

Comment:
Nit: "Revision of `EFI_PARTITION_INFO_PROTOCOL_REVISION`"

Reply:
done